### PR TITLE
Fix displayName for ES6 class component

### PR DESCRIPTION
**Pull request by havenchyk**
Here is a method how React understands component's name https://github.com/facebook/react/blob/b3c75d82b2c8862a461b34167bbef37a3eaa5c46/src/classic/element/ReactElementValidator.js#L68

I guess react-guard can use the same

### CODE REVIEW

File: test/react-15/index.js [hunk 1]
```diff
@@ -75,7 +74,6 @@ test.always.afterEach(() => {
     class PureClassComponent extends React.PureComponent {
       render () { throw {error: 'test'} }
     }
-    PureClassComponent.displayName = 'PureClassComponent'
     const result = ReactDOMServer.renderToStaticMarkup(React.createElement(PureClassComponent, {a: 'A'}, 'children'))
     t.is(result, '<div>w00t</div>')
     t.true(guardSpy.called)
```
Comment (kossnocorp) [inline test/react-15/index.js:125]: It's worth to add more examples. These lines are responsible for `this.constructor.displayName` test, but there is no tests for `this.constructor.name`

**Review by kossnocorp — CHANGES_REQUESTED** [review]
Thank you! It makes sense, but tests are missing.

**Comment by havenchyk** [comment]
Added some tests. Since property `name` of the constructor is read only, I just did `displayName` empty. /cc @kossnocorp

**Comment by havenchyk** [comment]
@kossnocorp just a friendly ping :)

**Review by kossnocorp — APPROVED** [review]

**Comment by kossnocorp** [comment]
Thank you! The update was released as `0.4.0`:

```
$ npm publish
+ react-guard@0.4.0
```

**Comment by havenchyk** [comment]
awesome!

**Pull request merged**